src/index/vocabulary/PolymorphicVocabulary.h

```diff
@@ -87,6 +88,39 @@ class PolymorphicVocabulary {
         },
         vocab_);
   }
+
+  // Analogous to `lower_bound`, but since `word` is guaranteed to be a full
+  // word, not a prefix, this function can respect the split of an underlying
+  // `SplitVocabulary`.
+  template <typename String, typename Comp, bool getUpperBound>
+  WordAndIndex getPositionOfWord(const String& word, Comp comp) const {
+    return std::visit(
+        [&word, &comp](auto& vocab) {
+          using T = std::decay_t<decltype(vocab)>;
+          if constexpr (ad_utility::isInstantiation<T, SplitVocabulary>) {
+            return vocab
+                .template getPositionOfWord<String, Comp, getUpperBound>(
+                    word, std::move(comp));
```

Comment:
Conceptually, why does `SplitVocabulary` have a different signature? Couldn't you just also use `upper_bound` and `lower_bound` there, and let the class delegate it accordingly?
Or alternatively implement `getPositionOfWord` for all other vocabularies using  `upper_bound` and `lower_bound`?